### PR TITLE
React v0.14 compatibility

### PR DESCRIPTION
**Pull request by L42y**
start from React v0.14, react split into two packages, react and
react-dom.

ref: https://facebook.github.io/react/blog/2015/07/03/react-v0.14-beta-1.html#two-packages

### CODE REVIEW

**Comment by aabenoja** [comment]
:+1: Though I am confused why `beta1` instead of `rc1`.

**Comment by kasperp** [comment]
@L42y thanks that is great. I would prefer not merging this to master until 0.14 is released. We can make a 0.14 branch and merge it into that if there is a need for it.

What do think ?

**Comment by aabenoja** [comment]
I like the idea of a `0.14` branch and keeping master "clean".

**Comment by L42y** [comment]
> I would prefer not merging this to master until 0.14 is released. We can make a 0.14 branch and merge it into that if there is a need for it

Great idea

File: ngReact.js [hunk 1]
```diff
@@ -9,17 +9,17 @@
 (function (root, factory) {
   if (typeof module !== 'undefined' && module.exports) {
     // CommonJS
-    module.exports = factory(require('react'), require('angular'));
+    module.exports = factory(require('react'), require('react-dom'), require('angular'));
   } else if (typeof define === 'function' && define.amd) {
     // AMD
-    define(['react', 'angular'], function (react, angular) {
-      return (root.ngReact = factory(react, angular));
+    define(['react', 'react', 'angular'], function (react, reactDOM, angular) {
```
Comment (aabenoja) [inline ngReact.js]: Should that middle import be `react-dom`, rather than `react`?

**Comment by aabenoja** [comment]
@L42y The minified file needs to be generated as well (with `npm run build`). The tests are green, but there are a couple of deprecation errors involving `React.unmountComponentAtNode` in both the test files.

**Comment by kasperp** [comment]
I created a react-0.14 branch, you can submit a pull-request against that - thanks

File: ngReact.js (hunk 1 above)
Comment (L42y) [inline ngReact.js]: As i seen in https://fb.me/react-dom-0.14.0-rc1.js, they used `define(['react'])`, so we should still use `'react'`, but I never used AMD at before, maybe I'm wrong.

**Comment by kasperp** [comment]
Replaced by #107

**Pull request closed without merging**